### PR TITLE
Add ert-storage as extras to setup.py

### LINKED ISSUE #1509: Install ert-storage from PyPi
`ert-storage 0.1.1` is available on pypi, this can now be added to the setup.py, and the install step in the CI can be removed or changed to just `pip install ert-storage` if it should not be included in `install_requires`.

### PR DESCRIPTION
**Pull request by ManInFez**
resolves #1509

### CODE REVIEW

**Review by dotfloat — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -63,6 +63,11 @@ def package_files(directory):
         "prefect",
         "dask_jobqueue",
     ],
+    extras_require={
+        "storage": [
+            "ert-storage==0.1.1",
```
Comment (markusdregi) [inline setup.py:68]: Any particular reason for the very strict pinning of `ert-storage` version?

Comment (dotfloat) [reply setup.py:68]: As Storage is in major version 0, breaking changes happen constantly. This requires simultaneous PRs in ert, webviz-ert and ert-storage, and they'll need to be merged at once. Pinning the version allows us to develop on the three projects separately, without fear that we might break the tests in a different project.

Comment (ManInFez) [reply setup.py:68]: in semver before version 1. we can change api without changing major version

Comment (markusdregi) [reply setup.py:68]: Makes sense! I'm hoping we can make this into a slightly wider range over time, but I think it is fine to pin it down exactly for now.

Comment (ManInFez) [reply setup.py:68]: yes, as soon as we make ert-storage 1.0.0, we are adhering to strict semver policies and are able to use at major version :)

**Pull request merged**